test.cpp

```diff
@@ -44,5 +62,17 @@ TEST(EngineTestSuite, ProperGear) {
   EXPECT_EQ(engine.currentGear_, 1);
 }
 
+TEST_F(HybridCarTestSuite, stopEngineShouldCallStopOEngine) {
+  EXPECT_CALL(*m_engineMock_petrol_rawPtr, stop());
+  EXPECT_CALL(*m_engineMock_electric_rawPtr, stop());
+  m_sut.stop_engine();
+}
+
+TEST_F(HybridCarTestSuite, startEngineShouldCallStopOEngine) {
+  EXPECT_CALL(*m_engineMock_petrol_rawPtr, start());
+  EXPECT_CALL(*m_engineMock_electric_rawPtr, start());
+  m_sut.start_engine();
+}
+
 
 
```

Comment:
Electric car tests are still missing.

Reply:
Added tests. Still some tests missing But is late....  I will add them I promise,.. maybe even parametrized.

Reply:
For me it's good enough :)
I know that you can write other similar tests.